src/mdio/segy/_workers.py

```diff
@@ -51,71 +53,60 @@ def header_scan_worker(segy_file: SegyFile, trace_range: tuple[int, int]) -> Hea
     return cast("HeaderArray", trace_header)
 
 
-def trace_worker(
+def trace_worker_v1(  # noqa: PLR0913
     segy_file: SegyFile,
-    data_array: Array,
-    metadata_array: Array,
-    grid: Grid,
-    chunk_indices: tuple[slice, ...],
-) -> tuple[Any, ...] | None:
-    """Worker function for multi-process enabled blocked SEG-Y I/O.
-
-    Performance of `zarr.Array` writes is slow if data isn't aligned with chunk boundaries,
-    sacrificing sequential reads of SEG-Y files. This won't be an issue with SSDs or cloud.
-
-    It retrieves trace numbers from the grid and gathers the current chunk's SEG-Y trace indices.
-    Then, it fills a temporary array in memory and writes to the `zarr.Array` chunk. We take full
-    slices across the sample dimension since SEG-Y data isn't chunked, eliminating concern.
+    out_path: str,
+    data_variable_name: str,
+    region: dict[str, slice],
+    grid_map: zarr_Array,
+    dataset: xr_Dataset,
+) -> SummaryStatistics | None:
+    """Writes a subset of traces from a region of the dataset of Zarr file.
 
     Args:
         segy_file: SegyFile instance.
-        data_array: Handle for zarr.Array we are writing traces to
-        metadata_array: Handle for zarr.Array we are writing trace headers
-        grid: mdio.Grid instance
-        chunk_indices: Tuple consisting of the chunk slice indices for each dimension
+        out_path: Output path for the Zarr file.
+        data_variable_name: Name of the data variable to write.
+        region: Region of the dataset to write to.
+        grid_map: Zarr array mapping live traces to their positions in the dataset.
+        dataset: Xarray dataset containing the data to write.
 
     Returns:
-        Partial statistics for chunk, or None
+        SummaryStatistics object containing statistics about the written traces.
     """
-    # Special case where there are no traces inside chunk.
-    live_subset = grid.live_mask[chunk_indices[:-1]]
-
-    if np.count_nonzero(live_subset) == 0:
+    if dataset.trace_mask.sum() == 0:
```

Comment:
```suggestion
    if not dataset.trace_mask.any():
```
I believe this will be a more efficient operation. Could you please double check this assumption?